library-client-rtsp/src/main/java/com/alexvas/rtsp/codec/FrameQueue.kt

```diff
@@ -21,19 +21,30 @@ class FrameQueue(frameQueueSize: Int) {
         if (queue.offer(frame, 5, TimeUnit.MILLISECONDS)) {
             return true
         }
+        if (queue.size >= frameQueueSize) {
+            Log.w(TAG, "Cannot add frame, queue is full. Queue size: ${queue.size}")
+            Thread.currentThread().interrupt()
+            queue.poll()
```

Comment:
Why do we make poll() if next call is clear()? poll() seems to be not needed

Reply:
When the queue reaches its maximum capacity, the behavior observed is that the frame at the oldest position becomes frozen. Interestingly, simply clearing the entire queue does not resolve the freezing issue. When we utilize the `poll()` method, it effectively resolves the issue of freezing.

Reply:
@alexeyvasilyev